test/powershell/Language/Parser/Parser.Tests.ps1

```diff
@@ -1,918 +1,918 @@
+    }
+
+	It "This test checks that we can use a range for globbing: [1-2] (line 557)" {
+        $result = ExecuteCommand "get-childitem $TestDrive/dir1\dir2\testdirfile[1-2].txt"
+		$result.Count | should be 2
+		$result[0].Name | should be "testdirfile1.txt"
+		$result[1].Name | should be "testdirfile2.txt"
+    }
+
+	It "This test will check that escaping the $ sigil inside single quotes simply returns the $ character. (line 583)" {
+        $result = ExecuteCommand "'`$'"
+		$result | should be "`$"
+    }
+
+	It "Test that escaping a space just returns that space. (line 593)" {
+        $result = ExecuteCommand '"foo` bar"'
+		$result | should be "foo bar"
+    }
+
+    It "Test that escaping the character 'e' returns the ESC character (0x1b)." {
+        $result = ExecuteCommand '"`e"'
+        $result | should be ([char]0x1b)
+    }
+
+    Context "Test Unicode escape sequences." {
+        # These tests require the file to be saved with a BOM.  Unfortunately when this UTF8 file is read by
+        # PowerShell without a BOM, the file is incorrectly interpreted as ASCII.
+        It 'Test that the bracketed Unicode escape sequence `u{0} returns minimum char.' {
+            $result = ExecuteCommand '"`u{0}"'
+            [int]$result[0] | should be 0
+        }
+
+        It 'Test that the bracketed Unicode escape sequence `u{10FFFF} returns maximum surrogate char pair.' {
+            $result = ExecuteCommand '"`u{10FFFF}"'
+            [int]$result[0] | should be 0xDBFF # max value for high surrogate of surrogate pair
+            [int]$result[1] | should be 0xDFFF # max value for low surrogate of surrogate pair
+        }
+
+        It 'Test that the bracketed Unicode escape sequence `u{a9} returns the © character.' {
+            $result = ExecuteCommand '"`u{a9}"'
+            $result | should be '©'
+        }
+
+        It 'Test that Unicode escape sequence `u{2195} in string returns the ↕ character.' {
+            $result = ExecuteCommand '"foo`u{2195}abc"'
+            $result | should be "foo↕abc"
+        }
+
+        It 'Test that the bracketed Unicode escape sequence `u{1f44d} returns surrogate pair for emoji 👍 character.' {
+            $result = ExecuteCommand '"`u{1f44d}"'
+            $result | should be "👍"
+        }
+
+        It 'Test that Unicode escape sequence `u{2195} in here string returns the ↕ character.' {
+            $result = ExecuteCommand ("@`"`n`n" + 'foo`u{2195}abc' + "`n`n`"@")
+            $result | should be "`nfoo↕abc`n"
+        }
+
+        It 'Test that Unicode escape sequence in single quoted is not processed.' {
+            $result = ExecuteCommand '''foo`u{2195}abc'''
+            $result | should be 'foo`u{2195}abc'
+        }
+
+        It 'Test that Unicode escape sequence in single quoted here string is not processed.' {
+            $result = ExecuteCommand @"
+@'
+
+foo``u{2195}abc
+
+'@
+"@
+            $result | should match "\r?\nfoo``u\{2195\}abc\r?\n"
```

Comment:
Actually, one question about `\r?\n` here. Is it because we use CRLF on windows and LF on Unix? If so, can we do the following instead?
```
$newLine = [System.Environment]::NewLine
$result | should be "${newLine}foo``u\{2195\}abc${newLine}"
```

Reply:
Yes, it is because git converts files to CRLF on Windows.  I like your suggestion and will make the change.